File: 287-find-the-duplicate-number/find-the-duplicate-number.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        n = len(nums)

        freq = [0] * n

        for i in range(len(nums)):
            freq[nums[i]] += 1

        for i in range(len(freq)):
            if freq[i] >= 2:
                return i

        return 0

File: 287-find-the-duplicate-number/find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int n = nums.size();

        vector<int>res(n+1, 0);

        for(int i = 0 ; i < n; i++){
            if(res[nums[i]] == 1){
                return nums[i];
            }

            res[nums[i]]++;
        }

        return 0;
    }
};